File: backend/src/main/java/com/go_exchange_easier/backend/dto/search/GetUserByNickResponse.java
package com.go_exchange_easier.backend.dto.search;

public record GetUserByNickResponse(

        Integer id,
        String nick,
        CountryDto countryOfOrigin,
        UniversityDto homeUniversity

) {

    public record UniversityDto(

            Short id,
            String nativeName,
            String englishName

    ) {}


    public record CountryDto(

            Short id,
            String name

    ) {}

}

File: frontend/src/dtos/search/GetUserByNickResponse.ts
import type { PageResponse } from "../common/PageResponse";

export type GetUserByNickResponse = PageResponse<{
  id: number;
  nick: string;
  countryOfOrigin: {
    id: number;
    name: string;
  } | null;
  homeUniversity: {
    id: number;
    nativeName: string;
    englishName: string;
  } | null;
}>;
